### PR TITLE
planner: prune columns again after predicate push down

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve? 

do another `columnPruner` rule after predicate push down.

### What is changed and how it works?

do another `columnPruner` rule after predicate push down.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Possible performance regression

### CODE REVIEW

File: planner/core/testdata/plan_suite_out.json [hunk 1]
```diff
@@ -1195,17 +1195,17 @@
       },
       {
         "SQL": "select /*+ HASH_AGG() */ t1.a from t t1 where t1.a < any(select t2.b from t t2)",
-        "Best": "LeftHashJoin{IndexReader(Index(t.f)[[NULL,+inf]]->Sel([if(isnull(test.t.a), <nil>, 1)]))->TableReader(Table(t)->HashAgg)->HashAgg->Sel([ne(Column#27, 0)])}->Projection->Projection",
```
Comment (SunRunAway) [inline planner/core/testdata/plan_suite_out.json]: After the second column pruning, the extra physical project can be eliminate in `postOptimize`.

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -374,7 +374,7 @@ func NewPlanBuilder(sctx sessionctx.Context, is infoschema.InfoSchema, processor
 
 // Build builds the ast node to a Plan.
 func (b *PlanBuilder) Build(ctx context.Context, node ast.Node) (Plan, error) {
-	b.optFlag = flagPrunColumns
+	b.optFlag = flagPrunColumns | flagPrunColumnsAgain
```
Comment (lzmhhh123) [inline planner/core/planbuilder.go]: Can we move `flagPrunColumnsAgain` to where assigned `flagPredicatePushDown`?

File: expression/schema.go [hunk 3]
```diff
@@ -179,6 +179,16 @@ func MergeSchema(lSchema, rSchema *Schema) *Schema {
 	return ret
 }
 
+// GetUsedList shows whether each column in schema is contained in usedCols.
+func GetUsedList(usedCols []*Column, schema *Schema) []bool {
```
Comment (lzmhhh123) [inline expression/schema.go]: Why do we move the function to package expression?

File: planner/core/planbuilder.go (hunk 2 above)
Comment (SunRunAway) [reply planner/core/planbuilder.go]: done.

File: expression/schema.go (hunk 3 above)
Comment (SunRunAway) [reply expression/schema.go]: It will be used by executor in future PR.

File: planner/core/planbuilder.go [hunk 4]
```diff
@@ -374,7 +374,12 @@ func NewPlanBuilder(sctx sessionctx.Context, is infoschema.InfoSchema, processor
 
 // Build builds the ast node to a Plan.
 func (b *PlanBuilder) Build(ctx context.Context, node ast.Node) (Plan, error) {
-	b.optFlag = flagPrunColumns
+	b.optFlag = flagPrunColumns // | flagPrunColumnsAgain
```
Comment (lzmhhh123) [inline planner/core/planbuilder.go]: Please don't use comment prefix to delete it. 😂

Comment (SunRunAway) [reply planner/core/planbuilder.go]: Sorry, fixed.

**Pull request merged**